src/internal/connector/exchange/attachment.go

```diff
@@ -61,6 +61,46 @@ func uploadAttachment(ctx context.Context, service graph.Service, userID, folder
 	return uploadLargeAttachment(ctx, service, userID, folderID, messageID, attachment)
 }
 
+func uploadEventAttachment(ctx context.Context, service graph.Service, userID, calendardID, eventID string,
```

Comment:
it seems like the event attachment upload functions are pretty near copies of the email attachment upload code, with the exception of sending the actual upload for small attachments/getting the upload URL. Due to that, it seems like it would be possible to use interfaces and more generic functions instead of duplicating code. For example, an interface with the following would cover the non-generic parts of the code I think:
```go
type attachmentUploadable interface {
  uploadSmallAttachment(ctx context.Context, attachment models.Attachmentable) error
  uploadSession(ctx context.Context) (models.UploadSessionable, error)
}
```

The implementations of this interface could internally keep the userID, calendar/folder ID, item IDs (so it's more flexible for stuff that may or may not need all of them), and service. The parameters for `uploadLargeAttachment` and `attachmentWriter` would need adjusted to remove the IDs (they're now in the interface implementation) and take the interface as a parameter like:
```go
func uploadLargeAttachment(
  ctx context.Context,
  uploader attachmentUploadable,
  attachment models.Attachmentable,
) error
```

The code that calls these functions could do the following:
1. instantiate an instance of the interface implementation for the given category of data
2. call `uploadAttachment(ctx, uploader, attachment)`

Reply:
Interface added [9a41952](https://github.com/alcionai/corso/pull/1345/commits/9a419523b43d1039769066b15dd77f15ee4a30ac)

Updates to the attachment.go [fdbea27](https://github.com/alcionai/corso/pull/1345/commits/fdbea271682a6f92875986db544d4d69adcf0f07)